chat/chat.py

```diff
@@ -0,0 +1,41 @@
+import openai
+from config import AppConfig
+from abc import ABC, abstractmethod
+
+class Chat(ABC):
+    @abstractmethod
+    def generate_response(self, query: str, context: str = None) -> str:
+        """Generate a response based on the query and optional context."""
+        pass
+
+class ChatOpenAI(Chat):
+    
+    def __init__(self):
+        config = AppConfig()
+        openai.api_key = config.openai_api_key
+        self.history = []
+
+    def generate_response(self, query, context=None):
+        if not query:
+            return ""  
+        
+        if context:
+            full_input = f"Given a context: {context}\n Given a query: {query}\n Please answer query based on the given context"
+        else:
+            full_input = query
+        
+        self.history.append({"role": "user", "content": full_input})
```

Comment:
it is very important to add history. However, we need to maintain or store this history in a database

Reply:
True, right now it only works if the chat object is not destroyed and still in memory. The next task is to manage the chat history using a database so the chat history can persist in a hard storage.

Reply:
Yep, storing in a database will also make it able to differentiate between each user's conversation history